async-openai/src/types/embedding.rs

```diff
@@ -64,6 +64,17 @@ pub struct Embedding {
     pub embedding: Vec<f32>,
 }
 
+/// Represents an base64-encoded embedding vector returned by embedding endpoint.
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
+pub(crate) struct Base64Embedding {
+    /// The index of the embedding in the list of embeddings.
+    pub(crate) index: u32,

```

Comment:
Any reason that this is `pub(crate)` and not `pub`? 

Every other type exposes all the fields. May be this can be pub too if someone wants raw base64?

Reply:
My initial thought was to hide the base64 encoding stuff from the end user. To them `EncodingFormat::Base64` "just speeds up the data transfer".

Reply:
As is, this struct will never actually be returned to the end user, since we take care of conversion internally, so it made sense to me to make it `pub(crate)`. I think there are two approaches:

1. Make this `pub` and return the raw base64-encoded embeddings. Difficulty with this is, you end up with a sort of "overload", where part of the request param impacts the response type. You would either need two functions or return an enum to match over.
2. Hide completely the base64 stuff by removing the generic. The user never has to think about any of this.

Reply:
Thank you for elaborating. 

(2) makes sense, however, if user is choosing base64 response - the library should provide them exact response - which if they want can convert easily by `try_into()` that you have.  Hypothetically, if this PR doesn't exist user would still get same result - except the data transfer optimization that you have mentioned. 

Hence, (1) is favorable to get best of both world and gives user a choice - returns base64 response and lets them convert via try_into.  

For (1) you're right we need two functions - we already do this for stream and non-streaming counter parts for chat completion API (and other streaming apis).

Something like  `Embeddings::create` and `Embeddings::create_base64` to make it explicit.

Reply:
In addition, two functions also gives you leverage to make it a non-breaking change by having a new type name for base64 counterpart.

Reply:
Hey just checking in - if there's any feedback/followup about the discussion?

Reply:
Hey sorry I forgot about this. Will have a go tonight.

Reply:
Oh no worries, thank you for following up, please take your time